### PR TITLE
Pol 245 new resource table 7

### PR DESCRIPTION
**Pull request by KVivek03**
### Description

New Resource Table 7

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

**Comment by KVivek03** [comment]
Hi @cdwilhelm , in the file **operational/azure/azuread_group_sync/azuread_group_sync.pt**, previously detail template was containing the data along with one "if" condition which was printing after the data details, when condition was satisfied.
I tried doing that using two export blocks but getting the error as two export blocks are not allowed. Please guide me how to proceed further.
FYR copying the line which i need to export form the old policy:
detail_template <<-EOS
## Groups that Will Be Synchronized from AzureAD
| Name  | CMP Group Href  | AzureAD Object Id |
| :---: | :-------------: | :---------------: |
{{ range data.groupsToUpdate -}}
  | {{ .name }} | {{ .href }} | {{ .objectId }} |
{{ end -}}
{{ $create_count := len data.groupsToCreate }}
{{ if (gt $create_count 0) -}}
{{ range data.groupsToCreate -}}
<br />
<br />
### The following groups need to be created and have permissions assigned to them in Governance:
{{ .name }}
{{ end -}}
{{ end -}}

PFB the approach i took(if no data will be there in the groupsToCreate array, nothing will be available for the 2nd export block):
export do
      resource_level true
      field "name" do
        label "Name"
        path "groupsToUpdate.name"
      end
      field "href" do
        label "CMP Group Href"
        path "groupsToUpdate.href"
      end
      field "objectId" do
        label "AzureAD Object Id"
        path "groupsToUpdate.objectId"
      end
      field "id" do
        label "Id"
        path "groupsToUpdate.id"
      end
    end
    export do
      resource_level true
      field "id" do
        label "Name"
        path "groupsToCreate.name"
      end
    end

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 1]
```diff
@@ -172,36 +172,44 @@ end
...
+        label "Subscription Name" 
+      end
+      field "name" do
+        label "Resource Name"
+      end
+      field "resource_group" do
+        label "Resource Group"
+      end
+      field "region" do
+        label "Azure Region"
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: the label should be Location.

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 2]
```diff
@@ -172,36 +172,44 @@ end
...
+      end
+      field "id" do
+        label "Resource ID"
+      end
+      field "tag" do
+        label "Tag Value"
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: should be Tags, and the value should be the list of tag key:value pairs

File: cost/azure/reserved_instances/recommendations/azure_reserved_instance_recommendations.pt [hunk 3]
```diff
@@ -135,28 +135,55 @@ script "js_ri_cleanup", type: "javascript" do
...
+    export do
+      resource_level true
+      field "skuName" do
+        label "SKU Name"
+      end
+      field "term" do
+        label "Term"
+      end
+      field "region" do
+        label "Region"
```
Comment (cdwilhelm) [inline cost/azure/reserved_instances/recommendations/azure_reserved_instance_recommendations.pt]: Should be Location

File: cost/billing_center_cost_anomaly/billing_center_cost_anomaly.pt [hunk 4]
```diff
@@ -392,35 +392,43 @@ script "js_format_costs", type: "javascript" do
...
     summary_template "Cost Anomaly Detected"
-    detail_template <<-EOS
-# Cost Anomaly Detected
-
-## Billing Centers: {{ data.billingCenters }}
-### Bill Currency: {{ data.currencyCode }}
-### Cost Metric: {{ parameters.param_cost_metric }}
-### Anomaly Threshold: {{ parameters.param_threshold }}
-### Time Period: {{ parameters.param_days }}
-### Minimum Spend for Period:  {{ parameters.param_min_spend}}
```
Comment (cdwilhelm) [inline cost/billing_center_cost_anomaly/billing_center_cost_anomaly.pt]: These details should not be removed.

**Comment by cdwilhelm** [comment]
Regarding operational/azure/azuread_group_sync/azuread_group_sync.pt, I'll remove this one from the list to update.  You can remove it from your commit.

File: cost/billing_center_cost_anomaly/billing_center_cost_anomaly.pt (hunk 4 above)
Comment (KVivek03) [reply cost/billing_center_cost_anomaly/billing_center_cost_anomaly.pt]: added back the details

File: cost/azure/reserved_instances/recommendations/azure_reserved_instance_recommendations.pt (hunk 3 above)
Comment (KVivek03) [reply cost/azure/reserved_instances/recommendations/azure_reserved_instance_recommendations.pt]: changed to location

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt (hunk 2 above)
Comment (KVivek03) [reply compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: added the tags filed in the required format

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt (hunk 1 above)
Comment (KVivek03) [reply compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: changed the field to Location

**Comment by KVivek03** [comment]
Hi @cdwilhelm is there anything pending from my side to be taken care of?

**Comment by KVivek03** [comment]
Hi @cdwilhelm , There are policies in which few of the fields are only being used in the CWF, they were not included in the detail_template block. Upon testing the new selectable actions feature, @SheikSena  and I observed that fields not included in the export block are passed as null to the CWF, or we can say that only export block fields are visible to the CWF.
I have modified policies on the basis of that and included those fields in the export block which are required for the CWF to execute properly. Please review the same and confirm us to proceed and modify other new resource table incident files also. Thanks.
cc @chandravurubindi

**Comment by cdwilhelm** [comment]
> Hi @cdwilhelm , There are policies in which few of the fields are only being used in the CWF, they were not included in the detail_template block. Upon testing the new selectable actions feature, @SheikSena and I observed that fields not included in the export block are passed as null to the CWF, or we can say that only export block fields are visible to the CWF.
> I have modified policies on the basis of that and included those fields in the export block which are required for the CWF to execute properly. Please review the same and confirm us to proceed and modify other new resource table incident files also. Thanks.
> cc @chandravurubindi

Yes, this is correct.  @SheikSena  and I discussed this on Thrusday.

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 5]
```diff
@@ -155,7 +155,8 @@ script "js_filter_resources", type: "javascript" do
               namespace: namespace,
               resource_type: resource_type,
               tag: rg_tag_value,
-              all_tags: entire_tags
+              all_tags: entire_tags,
+			  tags_obj: tags_obj
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: please line up fields.

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 6]
```diff
@@ -168,8 +169,9 @@ script "js_filter_resources", type: "javascript" do
               resource_group: resource_group_name,
               namespace: namespace,
               resource_type: resource_type,
-              tag: 'N/A',
-              all_tags: entire_tags
+              tag: 'none',
+              all_tags: entire_tags,
+			  tags_obj: tags_obj
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: please line up fields

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt (hunk 5 above)
Comment (KVivek03) [reply compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: corrected indentation

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt (hunk 6 above)
Comment (KVivek03) [reply compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: corrected indentation

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 7]
```diff
@@ -36,6 +36,13 @@ parameter "param_tag_key" do
   EOS
 end 
 
+parameter "param_automatic_tag" do
+    type "string"
+    label "Automatically Tag resources"
+    description "When this value is set to yes this policy will automatically take the tag resources action"
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt:42]: consider changing to:
When this value is set to yes this policy will automatically tag the resources.

File: cost/azure/rightsize_sql_instances/azure_rightsize_sql_instances.pt [hunk 8]
```diff
@@ -46,6 +46,13 @@ parameter "param_email" do
   label "Email addresses"
   description "Email addresses of the recipients you wish to notify"
 end
+parameter "param_automatic_rightsize_instance" do
+    type "string"
+    label "Automatically rightsize sql instance"
+    description "When this value is set to yes this policy will automatically take the rightsize sql instance action"
```
Comment (cdwilhelm) [inline cost/azure/rightsize_sql_instances/azure_rightsize_sql_instances.pt:52]: Consider changing to:
When this value is set to yes this policy will automatically rightsize the sql instance

File: cost/azure/unused_sql_databases/azure_unused_sql_databases.pt [hunk 9]
```diff
@@ -29,6 +29,14 @@ parameter "param_exclusion_tag_key" do
   description "Azure-native SQL Database tag key to ignore databases. Only supply the tag key. The policy assumes that the tag value is irrelevant."
 end
 
+parameter "param_automatic_delete_db" do
+    type "string"
+    label "Automatically delete unused sql database"
+    description "When this value is set to yes this policy will automatically take the delete unused sql database action"
```
Comment (cdwilhelm) [inline cost/azure/unused_sql_databases/azure_unused_sql_databases.pt:35]: Consider:
When this value is set to yes this policy will automatically delete the unused sql database

**Pull request merged**